### ISSUE TITLE
getIncomers doesn't consider egdes of the format nodeid__handlerid

### ISSUE THREAD

**Issue by pSnehanshu** [wbkd/react-flow#591]
The current `getIncomers` doesn't take into account edges incoming from custom nodes having multiple Handlers. These handlers have ids of the format `nodeid__handlerid`.

The following is a re-write of the same function, I have included a regex tester to include them.

```javascript
const getIncomers = (node, elements) => {
  if (!isNode(node)) {
    return [];
  }

  const incomersIds = elements
    .filter((e) => isEdge(e) && e.target === node.id)
    .map((e) => e.source);

  return elements.filter((e) => incomersIds.map((id) => {
    const matches = /([\w-^]+)__([\w-]+)/.exec(id);
    if (matches === null) {
      return id;
    }
    return matches[1];
  }).includes(e.id));
};
```

The buggy function was probably introduced [here](https://github.com/wbkd/react-flow/issues/514).

**Comment by chrtze** [comment]
This will be fixed in the next version. You can already use it with `npm install react-flow-renderer@next`. #657

**Comment by chrtze** [comment]
released in [7.0.0](https://github.com/wbkd/react-flow/releases/tag/7.0.0)
